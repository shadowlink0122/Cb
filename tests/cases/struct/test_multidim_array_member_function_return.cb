// 多次元配列の基本テスト

void main() {
    println("=== Test 1: Basic 2D array ===");
    
    int[2][3] matrix = [[1, 2, 3], [4, 5, 6]];
    
    for (int i = 0; i < 2; i++) {
        for (int j = 0; j < 3; j++) {
            println(matrix[i][j]);
        }
    }
    
    println("=== Test 2: 3x3 identity matrix ===");
    
    int[3][3] identity = [[1, 0, 0], [0, 1, 0], [0, 0, 1]];
    
    for (int i = 0; i < 3; i++) {
        for (int j = 0; j < 3; j++) {
            println(identity[i][j]);
        }
    }
    
    println("=== All tests passed ===");
}
