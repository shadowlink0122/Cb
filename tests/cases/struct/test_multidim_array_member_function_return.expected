=== Test 1: Basic 2D array ===
1
2
3
4
5
6
=== Test 2: 3x3 identity matrix ===
1
0
0
0
1
0
0
0
1
=== All tests passed ===
